RadarSDK/RadarAPIClient.m

```diff
@@ -389,4 +391,109 @@ - (void)searchGeofencesWithLocation:(CLLocation * _Nonnull)location
+        @"User-Agent": [RadarUtils userAgent],
+        @"X-Radar-Config": @"true",
+    };
+
+    [self.apiHelper requestWithMethod:@"GET" url:url headers:headers params:nil completionHandler:^(RadarStatus status, NSDictionary * _Nullable res) {
+        if (status != RadarStatusSuccess || !res) {
+            return completionHandler(status, nil, nil);
+        }
+
+        id addressesObj = res[@"addresses"];
+        NSArray<RadarAddress *> *addresses = [RadarAddress addressesFromObject:addressesObj];
+        if (addresses) {
+            return completionHandler(RadarStatusSuccess, res, addresses);
+        }
+
+        completionHandler(RadarStatusErrorServer, nil, nil);
+    }];
+}
+
+- (void)reverseGeocode:(CLLocation *)location completionHandler:(RadarGeocodeCompletionHandler)completionHandler {
+    NSString *publishableKey = [RadarSettings publishableKey];
+    if (!publishableKey) {
+        return completionHandler(RadarStatusErrorPublishableKey, nil, nil);
+    }
+
+    NSMutableString *qs = [NSMutableString new];
+    [qs appendFormat:@"latitude=%.15f", location.coordinate.latitude];
+    [qs appendFormat:@"&longitude=%.15f", location.coordinate.longitude];
+
+    NSString *host = [RadarSettings host];
+    NSString *url = [NSString stringWithFormat:@"%@/v1/geocode/reverse?%@", host, qs];
+    url = [url stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet URLQueryAllowedCharacterSet]];
+
+    NSDictionary *headers = @{
+        @"Authorization": publishableKey,
+        @"Content-Type": @"application/json",
+        @"User-Agent": [RadarUtils userAgent],
+        @"X-Radar-Config": @"true",
+    };
+
+    [self.apiHelper requestWithMethod:@"GET" url:url headers:headers params:nil completionHandler:^(RadarStatus status, NSDictionary * _Nullable res) {
+        if (status != RadarStatusSuccess || !res) {
+            return completionHandler(status, nil, nil);
+        }
+
+        id addressesObj = res[@"addresses"];
+        NSArray<RadarAddress *> *addresses = [RadarAddress addressesFromObject:addressesObj];
+        if (addresses) {
+            return completionHandler(RadarStatusSuccess, res, addresses);
+        }
+
+        completionHandler(RadarStatusErrorServer, nil, nil);
+    }];
+}
+
+- (void)ipGeocode:(RadarIPGeocodeCompletionHandler)completionHandler {
+    NSString *publishableKey = [RadarSettings publishableKey];
+    if (!publishableKey) {
+        return completionHandler(RadarStatusErrorPublishableKey, nil, nil);
+    }
+
+    NSString *host = [RadarSettings host];
+    NSString *url = [NSString stringWithFormat:@"%@/v1/geocode/ip", host];
+    url = [url stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet URLQueryAllowedCharacterSet]];
+
+    NSDictionary *headers = @{
+        @"Authorization": publishableKey,
+        @"Content-Type": @"application/json",
+        @"User-Agent": [RadarUtils userAgent],
+        @"X-Radar-Config": @"true",
+    };
```

Comment:
are these the standard headers for our request? thinking if we shall consolidate that to an util method

Reply:
definitely a good idea, but it seems like there are certain instances where we omit the `X-Radar-Config` key -- https://github.com/radarlabs/radar-sdk-ios/blob/v3/RadarSDK/RadarAPIClient.m#L257-L261

Nick, no rush, but could you provide more insight here? This seems like the `verifyEvent` path, which I don't think gets much usage..